components/SingleSelectRadio/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { ComponentContainer, RadioInput, RadioLabel } from './styles';
+
+interface Option {
+  label: string;
+  value: string;
+}
+
+interface SingleSelectRadioGroupProps {
+  name: string;
+  options: Option[];
+  defaultValue?: string;
+  onChange?: (value: string) => void;
+}
+
+export default function SingleSelectRadioGroup({
+  name,
+  options,
+  defaultValue,
+  onChange,
+}: SingleSelectRadioGroupProps) {
+  const [selectedValue, setSelectedValue] = useState(
+    defaultValue || options[0].value,
+  );
+
+  const handleChange = (value: string) => {
+    setSelectedValue(value);
+    if (onChange) onChange(value);
+  };
+
+  return (
+    <div>
+      {options.map(option => (
+        <ComponentContainer
+          key={option.value}
+          isSelected={selectedValue === option.value}
+        >
+          <RadioLabel isSelected={selectedValue === option.value}>
```

Comment:
can you add `htmlFor={option.label + 'Radio'}` here and id={option.label + 'Radio'} in `RadioInput` below so that the text is also clickable / is connected to the radio? 

Some Context (from chatgpt):
* Purpose: The id attribute is a global HTML attribute used to uniquely identify a single HTML element on a page.
* Usage: Primarily used to link labels with form inputs, create anchor links, or for CSS styling and JavaScript targeting.
* Requirement: Each id must be unique across the entire HTML document, as duplicate ids can cause issues with accessibility, JavaScript functionality, and CSS selectors.
* Accessibility: id is used by screen readers and can be useful for linking elements together for accessibility purposes (like associating <label> with an <input>).

i.e. it should look like this 

```
<ComponentContainer
          key={option.value}
          isSelected={selectedValue === option.value}
          onClick={() => handleChange(option.value)}
        >
          <RadioLabel isSelected={selectedValue === option.value} htmlFor={option.label + 'Radio'}>
            {option.label}
          </RadioLabel>
          <RadioInput
            name={name}
            value={option.value}
            checked={selectedValue === option.value}
            id={option.label + 'Radio'}
          />
</ComponentContainer>
```